script.js

```diff
@@ -64,27 +82,124 @@ const createCardDiv = (card) => {
   cardDiv.appendChild(backCardDiv);
   return cardDiv;
 };
-// flippingCards-----------------------------------------------------------------------------
-const flippingCards = (flipCard ) => {
-  flipCard .classList.toggle("flipped");
+
+// flippingCards-------------------------------------------------------------------
+
+const flippingCards = (flipCard) => {
+  flipCard.classList.add("flipped");
 };
-// rendering cards------------------------------------------------------------------------
+// flippingBack cars--------------------------------------------------------------
+
+let flipingBackCard = (flipBack) => {
+  flipBack.classList.remove("flipped");
+};
+// rendering cards-----------------------------------------------------------------
+
 const updateCards = () => {
   const shuffledcards = shufflingCards();
   const renderShuffledCards = () => {
     shuffledcards.map((card) => {
       const cardDiv = createCardDiv(card);
       cardContainer.appendChild(cardDiv);
       cardDiv.addEventListener("click", () => {
-        flippingCards(cardDiv);
+        clickHandker(cardDiv, card);
       });
     });
   };
   renderShuffledCards();
 };
 updateCards();
+
+// click handler function----------------------------------------------------
+
+function clickHandker(cardDiv, card) {
+  clickCounter++;
+  if (clickCounter === 1) {
+    startTimer();
+  }
+
+  flippingCards(cardDiv);
+
+  if (!matchedCards.includes(card) && !selectedCards.includes(card)) {
+    selectedCards.push(card);
+  }
+
+  if (selectedCards.length === 2) {
+    checkMatch();
+    movementsCounter++;
+    movementsContainer.textContent = movementsCounter;
+  }
+}
+
+// checkMatch function--------------------------------------------------------
+
+function checkMatch() {
+  let firstCard = selectedCards[0];
+  let secondCard = selectedCards[1];
+  let firstCardDiv = document.getElementById(`${firstCard.id}`);
+  let secondCardDiv = document.getElementById(`${secondCard.id}`);
+
+  if (firstCard.name === secondCard.name) {
+    firstCardDiv.classList.add("flipped");
+    secondCardDiv.classList.add("flipped");
+    selectedCards = [];
+
+    matchedCards.push(firstCard);
+    matchedCards.push(secondCard);
+    
+    // check if all cards matched
+    if(matchedCards.length === 16){
+      stopTimer()
+      alert('you win')
+    }
+    
+  } else {
+    setTimeout(() => {
+      flipingBackCard(firstCardDiv);
+      flipingBackCard(secondCardDiv);
+    }, 1000);
+    selectedCards = [];
+  }
+}
+
+// start timer function------------------------------------------------
+
+const startTimer = () => {
+  let second = 0;
+  let minute = 0;
+  timer = setInterval(() => {
+    second++;
+    if (second >= 60) {
+      second = 0;
+      minute++;
+    }
+    seconds.textContent = second;
+    minutes.textContent = minute;
```

Comment:
It would be more conventional and look nicer if the time was displayed in a mm:ss format 🙏🏽

Reply:
done :)